modules/SerialCommHub/main/serial_communication_hubImpl.cpp

```diff
@@ -4,452 +4,160 @@
-
-    {
-        std::scoped_lock lock(serial_mutex);
-
-        // Write command
-        int bytes_written = serial_device->write(message.data(), message.size());
-        if (bytes_written != message.size()) {
-            status.status_code = types::serial_comm_hub_requests::StatusCodeEnum::TxFailed;
-            return status;
-        }
-        serial_device->drain();
-
-        // Receive reply with timeout
-        bytes_received = serial_device->read(receive_buffer, sizeof(receive_buffer));
-        last_message_end_time = std::chrono::steady_clock::now();
-    }
-
-    // check if timed out
-    if (bytes_received <= 0) {
-        status.status_code = types::serial_comm_hub_requests::StatusCodeEnum::RxTimeout;
-        return status;
-    }
-
-    // enough bytes received?
-    if (bytes_received < 3) {
-        status.status_code = types::serial_comm_hub_requests::StatusCodeEnum::RxBrokenReply;
-        return status;
-    }
-
-    // check if from correct device
-    if (target_device_id != receive_buffer[0]) {
-        status.status_code = types::serial_comm_hub_requests::StatusCodeEnum::RxWrongDevice;
-        return status;
-    }
-
-    // check if CRC is correct
-    if (add_crc16 && !check_crc(receive_buffer, bytes_received)) {
-        status.status_code = types::serial_comm_hub_requests::StatusCodeEnum::RxCRCMismatch;
-        return status;
-    }
-
-    // All checks succeeded, return message
-    status.status_code = types::serial_comm_hub_requests::StatusCodeEnum::Success;
-    status.value = std::vector<int>(receive_buffer + 1, receive_buffer + bytes_received - 2);
+types::serial_comm_hub_requests::Result
+serial_communication_hubImpl::handle_modbus_read_holding_registers(int& target_device_id, int& first_register_address,
+                                                                   int& num_registers_to_read) {
 
-    return status;
-}
-
-types::serial_comm_hub_requests::Result serial_communication_hubImpl::handle_modbus_read_holding_registers(
-    int& target_device_id, int& first_register_address, int& num_registers_to_read, int& pause_between_messages) {
-    std::vector<std::uint8_t> response;
     types::serial_comm_hub_requests::Result result;
-    result.status_code = types::serial_comm_hub_requests::StatusCodeEnum::ModbusException;
+    std::vector<uint16_t> response;
 
     {
         std::scoped_lock lock(serial_mutex);
-        auto time_since_last_message = std::chrono::steady_clock::now() - last_message_end_time;
-        auto pause_time = std::chrono::milliseconds(pause_between_messages);
-        if (time_since_last_message < pause_time)
-            std::this_thread::sleep_for(pause_time - time_since_last_message);
 
-        bool done{false};
         uint8_t retry_counter{this->num_resends_on_error};
-        do {
-            if (retry_counter < 1) {
-                return result;
-            }
+        while (retry_counter-- > 0) {
```

Comment:
only style issue,  readability is a bit harder because of `(retry_counter-- > 0)` is not the same as `(--retry_counter > 0)`, therefor I prefer to put the `retry_counter--` into the while loop itself.
I guess you could also write `auto retry_counter = this->num_resends_on_error;`